### PR TITLE
servlets: request recognition

### PR DESCRIPTION
**Pull request by slavidichkov**

### CODE REVIEW

File: src/main/java/com/clouway/task2/http/Recognition.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.Set;
+
+/**
+ * @author Slavi Dichkov (slavidichkof@gmail.com)
+ */
+public class Recognition extends HttpServlet {
```
Comment (tsony-tsonev) [inline src/main/java/com/clouway/task2/http/Recognition.java]: not a good name - PageRecognizer or something like that will be better

File: src/main/java/com/clouway/task2/server/Jetty.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        webapp.addEventListener(new ServletContextListener() {
+
+            public void contextInitialized(ServletContextEvent servletContextEvent) {
+                ServletContext servletContext = servletContextEvent.getServletContext();
+                Set<String> pages=new HashSet<String>(){{
+                    add("first");
+                    add("second");
+                    add("third");
+                }};
+                servletContext.addServlet("connector", new Recognition(pages)).addMapping("/connector");
```
Comment (tsony-tsonev) [inline src/main/java/com/clouway/task2/server/Jetty.java]: also better endpoint name should be used, cause /connector to be handled by Recognition doesn't sound good

File: src/test/java/com/clouway/task2/http/RecognitionTest.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        final ByteArrayOutputStream out = new ByteArrayOutputStream();
+        context.checking(new Expectations() {{
+            oneOf(request).getParameter("page");
+            will(returnValue("second"));
+            oneOf(response).getWriter();
+            will(returnValue(new PrintWriter(out)));
+        }});
+        recognition.doGet(request,response);
+        String expected=out.toString();
+        assertThat(expected.contains("second page"),is(true));
```
Comment (mgenov) [inline src/test/java/com/clouway/task2/http/RecognitionTest.java]: There is a hamcrest matcher for String values. 

assertThat(expected, containsString("second page")); 


http://hamcrest.org/JavaHamcrest/javadoc/1.3/org/hamcrest/core/StringContains.html#containsString(java.lang.String)

File: src/main/java/com/clouway/task2/http/Recognition.java (hunk 1 above)
Comment (slavidichkov) [inline src/main/java/com/clouway/task2/http/Recognition.java]: fixed

File: src/main/java/com/clouway/task2/server/Jetty.java (hunk 2 above)
Comment (slavidichkov) [inline src/main/java/com/clouway/task2/server/Jetty.java]: fixed

File: src/test/java/com/clouway/task2/http/RecognitionTest.java (hunk 3 above)
Comment (slavidichkov) [inline src/test/java/com/clouway/task2/http/RecognitionTest.java]: fixed

File: src/main/java/com/clouway/task2/http/Recognition.java (hunk 1 above)
Comment (tsony-tsonev) [inline src/main/java/com/clouway/task2/http/Recognition.java]: I meant that the word Recognition is bad for class name, cause recognition is the task that the class will do, so the class name should be PageRecognizer

**Comment by tsony-tsonev** [comment]
LGTM, merge it after the renaming.

File: src/main/java/com/clouway/task2/http/Recognition.java (hunk 1 above)
Comment (slavidichkov) [inline src/main/java/com/clouway/task2/http/Recognition.java]: fixed

**Pull request merged**